spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -948,6 +950,17 @@ public static org.apache.spark.sql.catalyst.TableIdentifier toV1TableIdentifier(
     return org.apache.spark.sql.catalyst.TableIdentifier.apply(table, database);
   }
 
+  static String baseTableUUID(org.apache.iceberg.Table table) {
+    if (table instanceof HasTableOperations) {
+      TableOperations ops = ((HasTableOperations) table).operations();
+      return ops.current().uuid();
+    } else if (table instanceof BaseMetadataTable) {
+      return ((BaseMetadataTable) table).table().operations().current().uuid();
+    } else {
+      throw new UnsupportedOperationException("Cannot fetch table operations for " + table.name());
```

Comment:
should this be replicated across all Spark versions? Also I would probably update the error msg to `Cannot retrieve UUID for table ...`

Reply:
done